File: 203-RemoveLinkedListElements/203-RemoveLinkedListElements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        dummy = ListNode(next = head)

        prev = dummy
        curr = head

        while curr:
            if curr.val == val:
                nxt = curr.next
                prev.next = nxt
                curr = nxt
            else:
                nxt = curr.next
                prev = curr
                curr = nxt
        
        return dummy.next
             

File: 203-RemoveLinkedListElements/203-RemoveLinkedListElements.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} val
 * @return {ListNode}
 */
var removeElements = function(head, val) {

    let dummy = new ListNode(-1, head)
    let prev = dummy
    let curr = head

    while (curr){
        if (curr.val === val){
            let nxt = curr.next
            prev.next = nxt
            curr = nxt
        }
        else{
            let nxt = curr.next
            prev = curr
            curr = nxt
        }
    }
    return dummy.next;
};